server/controllers/conversation.js

```diff
@@ -0,0 +1,113 @@
+const Conversation = require('../models/Conversation');
+const asyncHandler = require('express-async-handler');
+
+// @route POST /create-conversation/
+// @desc create a new conversation
+// @access Public
+
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const { description, receiver } = req.body;
+
+  if (!description || !receiver) {
+    res.status(400);
+    throw new Error('Bad request! Missing description or receiver!');
+  }
+
+  const conversation = await Conversation.create({
+    messages: {
+      sender: req.user.id,
+      receiver,
+      description,
+    },
+    participants: [req.user.id, receiver],
+  });
+
+  res.status(200).json({
+    success: {
+      conversation,
+    },
+  });
+});
+
+// @route GET /message/:conversationId
+// @desc get all messages from a single conversation
+// @access Public
+
+exports.getAllMessages = asyncHandler(async (req, res, next) => {
```

Comment:
At the moment we don't need this controller as the `getAllConversations` controller returns the messages within every conversation. However, if we take the approach of populating messages on demand, we can refactor and make use of this route

Reply:
Yes the way getAllConversations works is just populating each convo with messages then it has them on the frontend ready to render on the initial mount.